lib/bluetooth/ble_scan/Kconfig

```diff
@@ -0,0 +1,120 @@
+#
+# Copyright (c) 2025 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig BLE_SCAN
+	bool "BLE scan"
+	depends on SOFTDEVICE
+	select EXPERIMENTAL
+	help
+	  scan.
+
+if BLE_SCAN
+
+config BLE_SCAN_BUFFER
+	int "Scan buffer length"
+	default 31
+	help
+	  Data length for an advertising set.
+
+config BLE_SCAN_NAME_MAX_LEN
+	int "Scan name maximum length"
+	default 32
+	help
+	  Maximum size for the name to search in the advertisement report.
+
+config BLE_SCAN_SHORT_NAME_MAX_LEN
+	int "Scan short name maximum length"
+	default 32
+	help
+	  Maximum size of the short name to search for in the advertisement report
+
+config BLE_SCAN_NAME_CNT
+	int "Scan name count"
+	default 1
+	help
+	  Number of name filters.
+
+config BLE_SCAN_APPEARANCE_CNT
+	int "Scan appearance count"
+	default 0
+	help
+	  Number of appearance filters.
+
+config BLE_SCAN_ADDRESS_CNT
+	int "Scan address count"
+	default 0
+	help
+	  Number of address filters.
+
+config BLE_SCAN_SHORT_NAME_CNT
+	int "Scan short name count"
+	default 0
+	help
+	  Number of short name filters.
+
+config BLE_SCAN_UUID_CNT
+	int "Scan UUID count"
+	default 0
+	help
+	  Number of filters for UUIDs.
+
+config BLE_SCAN_SCAN_INTERVAL
```

Comment:
Personal thought: Scan_scan feels a bit weird, perhaps we can remove one? Should still be descriptive enough, at least with the prompt.